.pre-commit-config.yaml

```diff
@@ -31,35 +31,74 @@ repos:
 
   # Text file hooks
   - repo: https://github.com/igorshubovych/markdownlint-cli
-    rev: v0.33.0
+    rev: v0.34.0
     hooks:
       - id: markdownlint
         args:
           - --config=.mdl_config.yaml
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v3.0.0-alpha.6
+    rev: v3.0.0-alpha.9-for-vscode
     hooks:
       - id: prettier
   - repo: https://github.com/adrienverge/yamllint
-    rev: v1.30.0
+    rev: v1.31.0
     hooks:
       - id: yamllint
         args:
           - --strict
 
   # GitHub Actions hooks
   - repo: https://github.com/python-jsonschema/check-jsonschema
-    rev: 0.22.0
+    rev: 0.23.0
     hooks:
       - id: check-github-actions
       - id: check-github-workflows
 
   # pre-commit hooks
   - repo: https://github.com/pre-commit/pre-commit
-    rev: v3.2.1
+    rev: v3.3.1
     hooks:
       - id: validate_manifest
 
+  # Go hooks
+  - repo: https://github.com/TekWizely/pre-commit-golang
+    rev: v1.0.0-rc.1
+    hooks:
+    # Style Checkers
+      - id: go-lint
+      - id: go-critic
+    # StaticCheck
+      - id: go-staticcheck-mod
+      - id: go-staticcheck-pkg
+      - id: go-staticcheck-repo-mod
+      - id: go-staticcheck-repo-pkg
```

Comment:
In many of these blocks similar hooks are repeated like they are here.  I'm not clear as to whether or not this is actually necessary.

Reply:
This is based on the sample config: https://github.com/TekWizely/pre-commit-golang/blob/master/sample-config.yaml Should I keep only one?

Reply:
I'm not sure.  It's difficult to tell what the differences are based just on the README for that project.

Reply:
Yes, it should be safe to remove those extras.

Reply:
The documentation for what the hooks do can be found in the hook definition file: [.pre-commit-hooks.yaml](https://github.com/TekWizely/pre-commit-golang/blob/master/.pre-commit-hooks.yaml). Some examples:
```yaml
# ==============================================================================
# go-build-mod
#   * Folder-Based
#   * Recursive
#   * Targets first parent folder with a go.mod file
#   * Executes if any .go files modified
#   * Executes if go.mod modified
# ==============================================================================

# ==============================================================================
# go-build-pkg
#   * Folder-Based
#   * Targets folder containing staged file
#   * Executes if any .go files modified
# ==============================================================================

# ==============================================================================
# go-build-repo-mod
#   * Repo-Based
#   * Recursive
#   * Targets ALL folders with a go.mod file
#   * Executes if any .go files modified
#   * Executes if go.mod modified
# ==============================================================================

# ==============================================================================
# go-build-repo-pkg
#   * Repo-Based
#   * Recursive
#   * Executes if any .go files modified
# ==============================================================================
```

Reply:
@jasonodoom, how did you decide that `go-build-mod` was the right choice over, say, `go-build-repo-mod`.  The latter seems more inclusive to me.

Reply:
@jsf9k yes was making these changes before I saw your comment based on Nick sharing the location of the hook definitions. I replaced the file recursive with those that have repo based options.